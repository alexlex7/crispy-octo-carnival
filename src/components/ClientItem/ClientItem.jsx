import React from 'react';
import styles from './ClientItem.module.css';
import PropTypes from 'prop-types';

export default function ClientItem({ client, isMobile = false }) {
  const { name, imageURL, mobileImageURL } = client;

  return (
    <li className={styles.listItem}>
      <img src={isMobile ? mobileImageURL : imageURL} alt={name} title={name} />
    </li>
  );
}

ClientItem.propTypes = {
  client: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    imageURL: PropTypes.string.isRequired,
    mobileImageURL: PropTypes.string.isRequired,
  }).isRequired,
  isMobile: PropTypes.bool,
};
