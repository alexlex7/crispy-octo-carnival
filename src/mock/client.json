{
  "id": "1",
  "name": "Mastercard",
  "imageURL": "http://localhost:8080/images/clients/masterCard.jpg",
  "mobileImageURL": "http://localhost:8080/images/clients/masterCardMobile.jpg"
}
