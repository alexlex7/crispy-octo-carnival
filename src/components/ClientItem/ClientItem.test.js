import { render, screen } from '@testing-library/react';
import ClientItem from './ClientItem';
import client from '../../mock/client.json';

test('renders client item', () => {
  const { asFragment } = render(<ClientItem client={client} />);
  const linkElement = screen.getByRole('img', { description: client.name });
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
