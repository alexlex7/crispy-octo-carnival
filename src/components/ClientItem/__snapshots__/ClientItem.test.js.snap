// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders client item 1`] = `
<DocumentFragment>
  <li
    class="listItem"
  >
    <img
      alt="Mastercard"
      src="http://localhost:8080/images/clients/masterCard.jpg"
      title="Mastercard"
    />
  </li>
</DocumentFragment>
`;
